src/bot/routes/rooms.rs

```diff
@@ -0,0 +1,186 @@
+use actix_multipart::Multipart;
+use actix_web::ResponseError;
+use actix_web::{web, Responder};
+use anyhow::{Context, Error};
+use futures_util::StreamExt as _;
+use matrix_sdk::ruma::api::client::room::create_room::v3::CreationContent;
+use matrix_sdk::ruma::RoomId;
+use matrix_sdk::ruma::{
+    api::{self},
+    events::{
+        AnyInitialStateEvent, EmptyStateKey, InitialStateEvent,
+    },
+    serde::Raw,
+};
+use matrix_sdk::Client as MatrixClient;
+use reqwest::StatusCode;
+use ruma::events::room::avatar::RoomAvatarEventContent;
+use ruma::OwnedMxcUri;
+use tokio::fs::File;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use crate::domain::{CommunityData, CommunityMatrixId, Uri};
+use crate::utils::{error_chain_fmt, mxc_to_download_uri};
+
+#[derive(thiserror::Error)]
+pub enum ServerError {
+    #[error("{0}")]
+    ValidationError(String),
+    #[error(transparent)]
+    UnexpectedError(#[from] anyhow::Error),
+}
+
+impl std::fmt::Debug for ServerError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        error_chain_fmt(self, f)
+    }
+}
+
+impl ResponseError for ServerError {
+    fn status_code(&self) -> reqwest::StatusCode {
+        match self {
+            ServerError::ValidationError(_) => StatusCode::BAD_REQUEST,
+            ServerError::UnexpectedError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
+}
+
+#[tracing::instrument(
+    name = "Task to reate a room", 
+    skip(
+        form
+    ), 
+    fields(
+        space_name = %form.name,
+        space_industry = %form.industry
+    )
+)]
+pub async fn create(
+    form: web::Json<CommunityData>,
+    matrix_client: web::Data<MatrixClient>,
+) -> Result<impl Responder, ServerError> {
+    println!("{:?}", form);
+    let response = on_handle_create_room(form.0,&matrix_client)
+        .await
+        .context("Failed to create a room")?;
+
+    Ok(web::Json(CommunityMatrixId {
+        id: response
+    }))
+}
+
+#[tracing::instrument(name = "Create a room", skip(matrix_client))]
+pub async fn on_handle_create_room(
+    room: CommunityData, 
+    matrix_client: &MatrixClient,
+) -> Result<String, Error> {
+    let mut request = api::client::room::create_room::v3::Request::new();
+    request.name = Some(&room.name);
+    
+    let topic = format!("{}\nTAG_VIRTO: {}", 
```

Comment:
This topic will be visible in UIs like element no? can the industry be part of a state event in json or similar? or we can give it a "name" that feels less machine readable(e.g. `"🏢️:"`)